### PR TITLE
Add CM copy kernel source

### PR DESCRIPTION
**Pull request by onabiull**

### CODE REVIEW

File: _studio/mfx_lib/genx/CMakeLists.txt [hunk 1]
```diff
@@ -77,6 +77,33 @@ if(MFX_ENABLE_KERNELS)
   set( sources "" )
   set( sources.plus "" )
 
+
+  # copy kernels
```
Comment (dvrogozh) [inline _studio/mfx_lib/genx/CMakeLists.txt]: why you moved this up? in that way it is more difficult to track file renames. Please, restore prev. order of ASC/copy/MCTF sections.

File: _studio/mfx_lib/genx/CMakeLists.txt [hunk 2]
```diff
@@ -77,6 +77,33 @@ if(MFX_ENABLE_KERNELS)
...
+  list( APPEND sources
+    ${prefix}/genx_copy_kernel_gen8_isa.cpp
+    ${prefix}/genx_copy_kernel_gen9_isa.cpp
+    ${prefix}/genx_copy_kernel_gen10_isa.cpp    
+    ${prefix}/genx_copy_kernel_gen11_isa.cpp
+    ${prefix}/genx_copy_kernel_gen11lp_isa.cpp
+  )
+
+  # field copy
+  set( prefix ${MSDK_GENX_ROOT}/field_copy/isa )
```
Comment (dvrogozh) [inline _studio/mfx_lib/genx/CMakeLists.txt]: We don't publish sources for field_copy yet?

File: _studio/shared/include/cm_mem_copy.h [hunk 3]
```diff
@@ -23,13 +23,13 @@
 
 #include "mfxdefs.h"
 #include "mfxstructures.h"
-#include "skl_copy_kernel_genx_isa.h"
+#include "genx_copy_kernel_gen9_isa.h"
 #if !(defined(AS_VPP_PLUGIN) || defined(UNIFIED_PLUGIN) || defined(AS_H264LA_PLUGIN))
```
Comment (dvrogozh) [inline _studio/shared/include/cm_mem_copy.h]: What the crap ifdef is that?! WHY exclude only gen8?!

File: _studio/shared/src/cm_mem_copy.cpp [hunk 4]
```diff
@@ -2616,23 +2616,23 @@ mfxStatus CmCopyWrapper::InitializeSwapKernels(eMFXHWType hwtype)
 #if !(defined(AS_VPP_PLUGIN) || defined(UNIFIED_PLUGIN) || defined(AS_H264LA_PLUGIN))
```
Comment (dvrogozh) [inline _studio/shared/src/cm_mem_copy.cpp]: you are kidding me... and here we exclude all the kernels.

File: _studio/mfx_lib/genx/CMakeLists.txt (hunk 1 above)
Comment (onabiull) [reply _studio/mfx_lib/genx/CMakeLists.txt]: ok

File: _studio/mfx_lib/genx/CMakeLists.txt (hunk 2 above)
Comment (onabiull) [reply _studio/mfx_lib/genx/CMakeLists.txt]: not yet

File: _studio/shared/src/cm_mem_copy.cpp (hunk 4 above)
Comment (onabiull) [reply _studio/shared/src/cm_mem_copy.cpp]: None of them defined in library build. So exactly opposite is true. For now only AS_H264LA_PLUGIN part remained relevant.

File: _studio/mfx_lib/genx/CMakeLists.txt [hunk 5]
```diff
@@ -94,23 +95,31 @@ if(MFX_ENABLE_KERNELS)
...
+  # field copy
   set( prefix ${MSDK_GENX_ROOT}/field_copy/isa )
   list( APPEND sources
     ${prefix}/genx_fcopy_gen8_isa.cpp
     ${prefix}/genx_fcopy_gen9_isa.cpp
     ${prefix}/genx_fcopy_gen10_isa.cpp
     ${prefix}/genx_fcopy_gen11_isa.cpp
     ${prefix}/genx_fcopy_gen11lp_isa.cpp
-  )
+  )
```
Comment (dvrogozh) [inline _studio/mfx_lib/genx/CMakeLists.txt]: don't modify

File: _studio/mfx_lib/genx/CMakeLists.txt [hunk 6]
```diff
@@ -77,6 +77,7 @@ if(MFX_ENABLE_KERNELS)
   set( sources "" )
   set( sources.plus "" )
 
+
```
Comment (dvrogozh) [inline _studio/mfx_lib/genx/CMakeLists.txt]: don't modify

File: _studio/mfx_lib/genx/CMakeLists.txt [hunk 7]
```diff
@@ -94,23 +95,31 @@ if(MFX_ENABLE_KERNELS)
     )
   endif()
 
+  # copy kernels
```
Comment (dvrogozh) [inline _studio/mfx_lib/genx/CMakeLists.txt]: comment is useless. It is obvious from the name of the folder...

File: _studio/mfx_lib/genx/CMakeLists.txt [hunk 8]
```diff
@@ -94,23 +95,31 @@ if(MFX_ENABLE_KERNELS)
...
-    ${prefix}/genx_icllp_copy_isa.cpp
-    ${prefix}/genx_cnl_copy_isa.cpp
+    ${prefix}/genx_copy_kernel_gen8_isa.cpp
+    ${prefix}/genx_copy_kernel_gen9_isa.cpp
+    ${prefix}/genx_copy_kernel_gen10_isa.cpp    
+    ${prefix}/genx_copy_kernel_gen11_isa.cpp
+    ${prefix}/genx_copy_kernel_gen11lp_isa.cpp
   )
 
+  # field copy
```
Comment (dvrogozh) [inline _studio/mfx_lib/genx/CMakeLists.txt]: comment is useless. It is obvious from the name of the folder...

File: _studio/shared/src/cm_mem_copy.cpp (hunk 4 above)
Comment (dvrogozh) [reply _studio/shared/src/cm_mem_copy.cpp]: I am trying to say: why these ifdefs are not aligned with the same ifdefs to include header files?

File: _studio/mfx_lib/genx/CMakeLists.txt [hunk 9]
```diff
@@ -94,13 +94,19 @@ if(MFX_ENABLE_KERNELS)
...
   set( prefix ${MSDK_GENX_ROOT}/copy_kernels/isa )
   list( APPEND sources
-    ${prefix}/genx_cht_copy_isa.cpp
-    ${prefix}/genx_skl_copy_isa.cpp
-    ${prefix}/genx_icl_copy_isa.cpp
-    ${prefix}/genx_icllp_copy_isa.cpp
-    ${prefix}/genx_cnl_copy_isa.cpp
+    ${prefix}/genx_copy_kernel_gen8_isa.cpp
+    ${prefix}/genx_copy_kernel_gen9_isa.cpp
+    ${prefix}/genx_copy_kernel_gen10_isa.cpp
```
Comment (dvrogozh) [inline _studio/mfx_lib/genx/CMakeLists.txt]: remove tailing whitespaces....

File: _studio/shared/src/cm_mem_copy.cpp [hunk 10]
```diff
@@ -2613,26 +2613,24 @@ mfxStatus CmCopyWrapper::InitializeSwapKernels(eMFXHWType hwtype)
     switch (hwtype)
     {
 #ifdef MFX_ENABLE_KERNELS
-#if !(defined(AS_VPP_PLUGIN) || defined(UNIFIED_PLUGIN) || defined(AS_H264LA_PLUGIN))
```
Comment (dvrogozh) [inline _studio/shared/src/cm_mem_copy.cpp]: :). Buy-buy:).

File: _studio/shared/include/cm_mem_copy.h [hunk 11]
```diff
@@ -23,14 +23,13 @@
 
 #include "mfxdefs.h"
 #include "mfxstructures.h"
-#include "skl_copy_kernel_genx_isa.h"
-#if !(defined(AS_VPP_PLUGIN) || defined(UNIFIED_PLUGIN) || defined(AS_H264LA_PLUGIN))
-#include "cht_copy_kernel_genx_isa.h"
+#ifdef MFX_ENABLE_KERNELS
```
Comment (dvrogozh) [inline _studio/shared/include/cm_mem_copy.h]: Good point. I missed that.

**Pull request merged**